token/provider.go

```diff
@@ -140,6 +141,73 @@ func (p *ManagementServiceProvider) managementService(aNew bool, opts ...Service
 	return ms, nil
 }
 
+func (p *ManagementServiceProvider) normalize(opt *ServiceOptions) (*ServiceOptions, error) {
+	// lookup configurations
+	configs, err := p.tmsProvider.Configs()
```

Comment:
Maybe it's better to create another struct that implements the normaliser interface and pass the slice of normalisers.